test/integration/cni/pod_traffic_across_az_test.go

```diff
@@ -0,0 +1,305 @@
+package cni
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/integration/common"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	coreV1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+)
+
+// Tests pod networking across AZs. It similar to pod connectivity test, but launches a daemonset, so that
+// there is a pod on each node across AZs. It then tests connectivity between pods on different nodes across AZs.
+var _ = Describe("[STATIC_CANARY] test pod networking", FlakeAttempts(3), func() {
+
+	var (
+		err        error
+		serverPort int
+		protocol   string
+
+		// The command to run on server pods, to allow incoming
+		// connections for different traffic type
+		serverListenCmd []string
+		// Arguments to the server listen command
+		serverListenCmdArgs []string
+
+		// The function that generates command which will be sent from
+		// tester pod to receiver pod
+		testConnectionCommandFunc func(serverPod coreV1.Pod, port int) []string
+
+		// The functions re-inforces that the positive test is working as
+		// expected by creating a negative test command that should fail
+		testFailedConnectionCommandFunc func(serverPod coreV1.Pod, port int) []string
+
+		// Expected stdout from the exec command on testing connection
+		// from tester to server
+		testerExpectedStdOut string
+		// Expected stderr from the exec command on testing connection
+		// from tester to server
+		testerExpectedStdErr string
+
+		// Daemonset to run on node
+		testDaemonSet *v1.DaemonSet
+
+		// Map of Pods placed on the node
+		interfaceToPodList common.InterfaceTypeToPodList
+
+		// Map of AZ name, string to pod IP, string
+		azName    string
+		azToPodIP map[string]string
+		azToPod   map[string]coreV1.Pod
+	)
+
+	JustBeforeEach(func() {
+		By("authorizing security group ingress on instance security group")
+		err = f.CloudServices.EC2().
+			AuthorizeSecurityGroupIngress(instanceSecurityGroupID, protocol, serverPort, serverPort, "0.0.0.0/0")
+		Expect(err).ToNot(HaveOccurred())
+
+		By("authorizing security group egress on instance security group")
+		err = f.CloudServices.EC2().
+			AuthorizeSecurityGroupEgress(instanceSecurityGroupID, protocol, serverPort, serverPort, "0.0.0.0/0")
+		Expect(err).ToNot(HaveOccurred())
+
+		netcatContainer := manifest.
+			NewNetCatAlpineContainer(f.Options.TestImageRegistry).
+			Command(serverListenCmd).
+			Args(serverListenCmdArgs).
+			Build()
+
+		By("creating a server DaemonSet on primary node")
+
+		testDaemonSet = manifest.
```

Comment:
I think it would be less complexity to use node selectors instead of deploying busybox pods as a Daemonset and adding all of the daemonset infra above. We can choose which nodes to deploy pods on by AZ

Reply:
>  I think it would be less complexity to use node selectors instead of deploying busybox pods as a Daemonset and adding all of the daemonset infra above. We can choose which nodes to deploy pods on by AZ

It is *intentional* to choose Daemonset. Different regions have different AZs and we want to ensure that nodes across different AZs are accounted for to run this traffic test.

*daemonset* is the simplest and automated way to ensure that. If we had to with node selection and spinning, it wouldn't be as straight forward ensuring, what daemonset already ensures.

Reply:
I have addressed the review comments.

I think, the motivation for using daemonset came, when we looked at our requirements in a standalone away. Ensure cluster nodes and across AZs; and exercise test for covering AZ failure. 

This test script is currently placed in cni repo, but if the scope increases, we should plan move this into separate repo.

Keeing daemonset for now. Let's see what others think.